addons/point_of_sale/static/src/scss/pos.scss

```diff
@@ -834,7 +834,14 @@ td {
 .pos .rightpane-header .categories-header {
     display: flex;
     overflow-x: auto;
+}
+
+.pos .rightpane-header .categories-header-scrollbar-hidden {
```

Comment:
since the CSS is common, you could have the class `.categories-header` in both cases, and add an additional class when you want to hide the scrollbar